usaspending_api/etl/management/commands/load_transactions_in_delta.py

```diff
@@ -168,6 +193,35 @@ def prepare_spark(self):
         if spark_created_by_command:
             self.spark.stop()
 
+    def award_id_lookup_pre_delete(self):
+        """
+        Need to create a table to store the award ids corresponding to the transaction_unique_ids that are
+        about to be deleted.
+        """
+        # This needs to be done here because Spark SQL does not have a RETURNING clause.
+        # Also, this needs to be a table, not a view because the records will be deleted from the table,
+        #   so a view would not be able to retain the values after the delete.
+        # This table does not need to be an external table, though, because it will be dropped in the
+        #   post-delete code.  Will put table in temp DB, therefore.
+        destination_database = "temp"
+        self.logger.info(f"Creating database {destination_database}, if not already existing.")
+        self.spark.sql(f"CREATE DATABASE IF NOT EXISTS {destination_database};")
+        self.spark.sql(
+            f"""
+                CREATE OR REPLACE TABLE {destination_database}.possibly_modified_awards
```

Comment:
Could you store the results of this in a dataframe instead of a table, so we can avoid extra writes/deletes to S3?

Reply:
Actually, based on some tests I'm doing, a dataframe also seems to be dependent on the state of the underlying data. I'm thinking there's gotta be a way to "materialize" it though. Exploring now.